src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -897,14 +898,25 @@ int xocl_set_cma_bank(struct xocl_drm *drm_p, uint64_t base_addr, size_t ddr_ban
 	return xocl_addr_translator_enable_remap(drm_p->xdev, base_addr, ddr_bank_size);
 }
 
-int xocl_cleanup_mem(struct xocl_drm *drm_p)
+int xocl_cleanup_mem(struct xocl_drm *drm_p, uint32_t slot_id)
 {
 	int ret;
 	mutex_lock(&drm_p->mm_lock);
-	ret = xocl_cleanup_mem_nolock(drm_p);
+	ret = xocl_cleanup_mem_nolock(drm_p, slot_id);
 	mutex_unlock(&drm_p->mm_lock);
 	return ret;
 }
+int xocl_cleanup_mem_all(struct xocl_drm *drm_p)
```

Comment:
why not just hold the drm_p->mm_lock in xocl_cleanup_mem_all()? We don't have per-slot lock, right?

Reply:
Yes. We can do that. Instead of lock/unlock multiple time here we can do that once. 
Fixed accordingly. Thanks